test/HashingTests.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE TypeApplications #-}
+
+module HashingTests where
+
+import Cryptography.Sodium.Bindings.Hashing
+import Foreign (Ptr, mallocBytes, peekArray0)
+import qualified Foreign as C
+import Foreign.C (CInt (..), CSize (..), CUChar)
+import qualified Foreign.C.String as C
+import Test.Tasty
+import Test.Tasty.HUnit
+
+spec :: TestTree
+spec =
+  testGroup
+    "Hashing tests"
+    [ testCase "Test hashing a single message with a key" testHashingWithAKey,
+      testCase "Test hashing a single message with no key" testHashingWithNoKey
+    ]
+
+testHashingWithAKey :: Assertion
+testHashingWithAKey = do
```

Comment:
In this setting, I would recommend the use of `allocaBytes` instead of `mallocBytes`. You don't need the memory to live on past this function call, and as-current, you leak memory.